src/charm.py

```diff
@@ -293,8 +292,15 @@ def _get_redis_relation_data(self) -> typing.Tuple[str, int]:
         for redis_unit in self._stored.redis_relation:  # type: ignore
             # mypy fails to see that this is indexable
             redis_unit_data = self._stored.redis_relation[redis_unit]  # type: ignore
-            redis_hostname = redis_unit_data.get("hostname", "")  # type: ignore
-            redis_port = redis_unit_data.get("port", DEFAULT_REDIS_PORT)  # type: ignore
+            try:
+                redis_hostname = str(redis_unit_data.get("hostname"))  # type: ignore
```

Comment:
How about just `redis_hostname = str(redis_unit_data.get("hostname") or "")`? This would ensure it's either a valid value or "" I think?

Reply:
I don't think catching `Exception` is a great solution.
What do you want to protect against ?
I'm wondering if an earlier pydantic validation of the data isn't a better approach

Reply:
The redis lib is happy to put whatever in the stored state: https://github.com/canonical/redis-k8s-operator/blob/main/lib/charms/redis_k8s/v0/redis.py#L79

Do we know the interface for the databag elements ?

Reply:
Juju relation data values are always strings. I think it's okay to just use `redis_unit_data.get("hostname", "")`.

Reply:
Yes, you are right, sorry I have misread that, I thought it's reading from the relation.